CRM/Civicase/Hook/PostProcess/RedirectToCaseDetails.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Redirect to case details post process hook.
+ */
+class CRM_Civicase_Hook_PostProcess_RedirectToCaseDetails {
+
+  /**
+   * Redirects the user to the case details after creating the case.
+   *
+   * @param string $formName
+   *   The class name of the submitted form.
+   * @param object $form
+   *   The submitted form instance.
+   */
+  public function run($formName, $form) {
```

Comment:
Does this hook affect the normal case creation via the Add New Case on main menu too?
I think that redirects to the case details by default right via this hook : https://github.com/compucorp/uk.co.compucorp.civicase/blob/2877d6cfcc29caacdf0a4b70ff8b81033e7e1b26/CRM/Civicase/Hook/PageRun/ViewCasePageRedirect.php#L17 ?

If we remove the Page run hook, will this new hook added work for the add case form on the Nav Menu then and redirect to the case details?

Reply:
Did some investigation on `ViewCasePageRedirect.php`, both hooks redirect the user to the Angular Case Details page, but both do them on different events:

* `ViewCasePageRedirect.php` redirects the user when they access the original CRM's case details page.
* `RedirectToCaseDetails.php` redirects the user after creating the case. 

Originally, when creating a new case through the `New Case` menu item, we would use the Case form, and this was redirecting us to the "view case" CRM page, which in turn redirected us to the case details page because of the `ViewCasePageRedirect.php` hook. `RedirectToCaseDetails.php` removes this extra step and directly redirects the user to the case details page. We still need the `ViewCasePageRedirect.php` hook because users could try to access the old case URL directly and this will redirect them to the right path. Finally, we also need `RedirectToCaseDetails.php` because `ViewCasePageRedirect.php` does not work when creating cases from a modal.